tests/BasicSearchTests.FunctionalTests.Core/BaseFunctionalTests.cs

```diff
@@ -14,11 +14,16 @@ public class BaseFunctionalTests
         protected RetryHandler RetryHandler;
 
         public BaseFunctionalTests()
+            : this(EnvironmentSettings.SearchServiceBaseUrl)
+        {
+        }
+
+        public BaseFunctionalTests(string baseUrl)
         {
             // Arrange
             IgnoreCertificateValidationErrors();
             RetryHandler = new RetryHandler(new HttpClientHandler());
-            Client = new HttpClient(RetryHandler) { BaseAddress = new Uri(EnvironmentSettings.SearchServiceBaseUrl) };
+            Client = new HttpClient(RetryHandler) { BaseAddress = new Uri(baseUrl) };
```

Comment:
`Client` should be disposed by making the test class `IDisposable`. I think `RetryHandler` too

Reply:
Disposed both `Client` and `RetryHandler`